src/main/java/org/skriptlang/skript/bukkit/toolcomponent/elements/ExprToolCompRules.java

```diff
@@ -0,0 +1,138 @@
+package org.skriptlang.skript.bukkit.toolcomponent.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.bukkitutil.ItemUtils;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.slot.Slot;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.components.ToolComponent;
+import org.bukkit.inventory.meta.components.ToolComponent.ToolRule;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.toolcomponent.ToolRuleWrapper;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+
+@Name("Tool Component - Tool Rules")
+@Description("The tool rules of a tool component.")
+@Examples({
+	"set {_rules::*} to the tool rules of {_item}",
+	"",
+	"set {_component} to the tool component of {_tool}",
+	"set {_rules::*} to the tool rules of {_component}"
+})
+@RequiredPlugins("Minecraft 1.20.6+")
+@Since("INSERT VERSION")
+public class ExprToolCompRules extends PropertyExpression<Object, ToolRule> {
+
+	static {
+		Skript.registerExpression(ExprToolCompRules.class, ToolRule.class, ExpressionType.PROPERTY,
+			"[the] tool rules of %itemstacks/itemtypes/slots/toolcomponents%");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr(exprs[0]);
+		return true;
+	}
+
+	@Override
+	protected ToolRule @Nullable [] get(Event event, Object[] source) {
+		List<ToolRule> rules = new ArrayList<>();
+		for (Object object : getExpr().getArray(event)) {
+			if (object instanceof ToolComponent component) {
+				rules.addAll(component.getRules());
+			} else {
+				ItemStack itemStack = ItemUtils.asItemStack(object);
+				if (itemStack == null)
+					continue;
+				rules.addAll(itemStack.getItemMeta().getTool().getRules());
+			}
+		}
+		return rules.toArray(ToolRule[]::new);
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		return switch (mode) {
+			case SET, DELETE, ADD, REMOVE -> CollectionUtils.array(ToolRuleWrapper.class);
+			default -> null;
+		};
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		ToolRule[] rules = null;
+		if (delta != null)
+			rules = (ToolRule[]) delta;
+		ToolRule[] finalRules = rules;
+
+		Consumer<ToolComponent> changer = switch (mode) {
+			case SET -> component -> {
+				component.setRules(Arrays.stream(finalRules).toList());
+			};
+			case DELETE -> component -> {
+				component.setRules(new ArrayList<>());
+			};
```

Comment:
Might be easier to do `component.getRules().clear()`
*edit: docs do say it's mutability of the list isn't guaranteed, I still think this is better than a full overwrite which could possibly poorly interact with other plugins if the same list was cached and not retrieved*